pybop/costs/_likelihoods.py

```diff
@@ -111,9 +111,12 @@ def __init__(
         self._dsigma_scale = dsigma_scale
         self._logpi = -0.5 * self.n_data * np.log(2 * np.pi)
 
+        # Add sigma parameter, join with self.parameters, reapply transformations
         self.sigma = Parameters()
         self._add_sigma_parameters(sigma0)
         self.parameters.join(self.sigma)
+        self.transformation = self.parameters.construct_transformation()
+        self.set_fail_gradient()
```

Comment:
this line confused me a bit as I wondererd why you were setting the fail gradient again, but then I figured it was because you change the number of parameters. I feel like this should be done in the base cost class, perhaps as a setter on  `self.parameters` which checks if the number of params changes and if so updates the fail gradient